truffle/com.oracle.truffle.api.test/src/com/oracle/truffle/api/vm/EngineTest.java

```diff
@@ -117,4 +121,43 @@ public void wrappedAsArray() throws Exception {
         assertEquals(2, arr[1].intValue());
         assertEquals(3, arr[2].intValue());
     }
+
+    @Test
+    public void initializePolyglotEngineWithConfiguration() throws IOException {
+        PolyglotEngine vm = createBuilder().config("application/x-test-import-export-1", "cmd-line-args", new String[]{"1", "2"}).config("application/x-test-import-export-2", "hello", "world").build();
+        PolyglotEngine.Language language1 = vm.getLanguages().get("application/x-test-import-export-1");
+
+        assertNotNull("Lang1 found", language1);
+
+        Ctx ctx1 = language1.getGlobalObject().as(Ctx.class);
+        String[] args = (String[]) ctx1.env.getConfig().get("application/x-test-import-export-1").get("cmd-line-args");
+        assertNotNull("Founds args", args);
+
+        assertEquals("1", args[0]);
+        assertEquals("2", args[1]);
+
+        assertNull("Can't see settings for other language", ctx1.env.getConfig().get("application/x-test-import-export-2"));
```

Comment:
I don't think there is any use for two levels of the map. Just merge values for all mimetypes of a language together into one map.